(maint) Update clojure/test.check to match other projects

Magisus opened the pull request:
Jar-jar and puppetdb now use version 1.1.1 of `clojure.test/check`, so this library's pin to 0.9.0 causes dependency confusion in projects that depend on this and either of those. This commit updates the version used here to match the rest of the PE ecosystem.

Magisus committed:
(maint) Move clojure/test.check to test profile and update to 1.1.1

This commit moves the `clojure/test.check` dependency to the `test` lein
profile, since it is only used in tests. This will prevent it from being
picked up transitively by other projects that depend on `jdbc-util`.
Previously, this could sometimes cause ambiguous dependency resolution,
since other projects, notably jar-jar and puppetdb, also pull in
`clojure/test.check`, recently at a different version than this library.

In addition to moving it to a test dependency, this commit also updates
the test library to the version used by other projects in the PE
ecosystem.

nwolfe commented on project.clj:
```diff
@@ -8,7 +8,7 @@
   :dependencies [[org.clojure/clojure]
                  [org.clojure/java.jdbc]
-                 [org.clojure/test.check "0.9.0"]
+                 [org.clojure/test.check "1.1.1"]
```
Shouldn't this be in the test dependencies so it's not transitively picked up by other projects?

Then this whole situation goes away.

Magisus replied:
Probably, come to think of it. A thing that may not have been known when it was added 8 years ago 🤔 Happy to move it there if we think that's fine.

Magisus replied:
It does only seem to be used in tests (wanted to make sure it wasn't used in helpers for other libraries' tests). I'll move it.

Magisus committed:
(maint) Trigger GH lein testing on all changes

Magisus commented:
I added a thing to trigger tests on changes to `project.clj`, seems like if we're changing dependencies we should run tests.

Magisus commented:
The test failures appears to be related to this https://www.postgresql.org/message-id/tencent_BD6B866E25069560051D16A1EEA07A4F2908%40qq.com and seems legit. I think this is the offending code: https://github.com/puppetlabs/jdbc-util/blob/092001bda49bbcdf1d3215d714dc46975cf524a9/src/puppetlabs/jdbc_util/core.clj#L114-L116.

```
ERROR in (create-db!-test) (QueryExecutorImpl.java:2676)
Uncaught exception, not in assertion.
expected: nil
  actual: org.postgresql.util.PSQLException: ERROR: CREATE DATABASE cannot be executed within a pipeline
```

It was missed when we upgraded from Postgres 9 to 14 [here](https://github.com/puppetlabs/jdbc-util/commit/dff95198ff7aca566ed1e226c51e9f4c732f062c), because the tests are gated to only run on changes to `src/` and `test/`. In this PR I added `project.clj` to that list, but that wouldn't have caught this either. I downgraded back to 9 and the failure went away. This feels like a real bug to me, but is unrelated to the changes in this PR.

nwolfe commented:
I would remove the `paths` from GHA workflows so tests are ran when CI files change, which would've caught the issue introduced at https://github.com/puppetlabs/jdbc-util/commit/dff95198ff7aca566ed1e226c51e9f4c732f062c 

PRs are few and far between, and tests are quick enough, just run them all for any change.

To fix the issue now, we could try pinning to the latest image that _does_ work?

Magisus committed:
(maint) Pin to Postgres 14.6

Postgres 14.7 started enforcing a restriction where databases could not
be created or dropped as part of a pipeline of SQL commands. We are
doing that in our `create-db` function when the user that is designated
to be the DB owner is not part of the `db-owner` role or a superuser.
That code should be fixed, though the fact that this isn't causing a
problem in PE, which is currently using Postgres 14.13, likely means
this path is unused.

In the meantime, this commit pins CI for this repo to 14.6, where the
existing code is still allowed.

steveax commented:
If it's quick (an hour or two), I say we fix the `create-db!` bug, otherwise, let's ticket it and fix later.
